### DISCUSSION TITLE
Connection fails with `Unsupported Chains`

### DISCUSSION THREAD

**Discussion by thxvatsal** [WalletConnect/walletconnect-monorepo#2748] [Developer Support]
Hello fellas! :wave: 

I am a maintainer for multi chain wallet package [Wallet01](https://github.com/Huddle01/Wallet01).

I have been trying to migrate our Walletconnect Connector from v1 to v2 and I have been following the docs accordingly, although when I try to scan and connect my metamask mobile, or even rainbow mobile wallet, it is throwing this error (attached below). :point_down: 

We have been passing the chains in an array as mentioned on the docs, something like this `[1 ,4, 137]`.

Feel free to have a look at the code here at this [PR](https://github.com/Huddle01/Wallet01/pull/24) 

Thanks in advance!!

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/95976661/7e243099-c245-4c30-9687-fb3790b8f2d5)

**Comment by glitch-txs** [comment]
Try removing 4 is deprecated and most wallets won't support it or even all testnets

You can also add optionalChains to prevent this error

**Reply by thxvatsal** [reply]
Hi! Thank for reaching out!

I tried with only chainId `1`, but it still throws the same error.

**Reply by glitch-txs** [reply]
Woah that project looks very interesting! Btw, do you think you can create a minimal reproducible example of the issue? I can help you debug. (I've done a similar package but for EVM only)
